### ISSUE TITLE
$PATH not set correctly when onivim not launched from command line

### ISSUE THREAD

**Issue by yunti** [onivim/oni2#1161]
Version 0.3.0 (macos catalina)

When I load up a .re file in a bucklescript project I get a 'Could not run bsb" error:

<img width="401" alt="image" src="https://user-images.githubusercontent.com/1826271/71769934-ccd11400-2f1e-11ea-8bea-f6a4faf5c7ae.png">

Reason completion and laguage server features don't work because of this.

**Comment by bryphe** [comment]
Thanks for the issue, @yunti !

- Are you running from finder or the command line?
- Are you using `bash` or `zsh` as your default shell? I suspect is an issue of us not sending the proper `PATH` to the extension host

**Comment by yunti** [comment]
Thanks, fish is the default shell. I normally just launch from spotlight but from finder has the same issue. (didn't realise there was an oni command to launch from cmd line...)
- Yup a `PATH` issue makes sense, perhaps because I'm defaulting to fish..

**Comment by bryphe** [comment]
Ah ok, sounds like a `PATH` issue - we try and get the default from `bash`, but we need to make it configurable (or find a better way to find the default shell, automatically).

@CrossR recently added an 'Add to PATH' functionality in our command palette:
![image](https://user-images.githubusercontent.com/13532591/71771474-3e459e00-2ef0-11ea-8c8f-a839202751d8.png)

Was blocked by #1160 - but sounds like the workaround of deleting `keybindings.json` worked for you.

If you run from `fish`, the `SHELL` environment variable should be populated, and we should pull the `PATH` from the right place 🤞

**Comment by yunti** [comment]
Thanks, tried  the new add to path command you mentioned - works well :).  Launching from the cmd line now makes it work which is great news,( but launching from finder (or spotlight still has the issue - but no deal breaker with the workaround.). 

(changing the default shell to bash or zsh has no impact)

**Comment by yunti** [comment]
Just a follow up on this (I amended the title for clarity). The screenshots below show the $PATH not being set correctly when launched from eg spotlight on macos. Launching from the command line sets $PATH as expected.

Launched from spotlight:
<img width="634" alt="image" src="https://user-images.githubusercontent.com/1826271/78979318-88ec3d00-7b13-11ea-83c3-3c2d736517d3.png">

Oni launched via the cmd line:
<img width="634" alt="image" src="https://user-images.githubusercontent.com/1826271/78979365-a15c5780-7b13-11ea-9730-76f2b9e25826.png">

**Comment by DanielPower** [comment]
Relates to #1548

**Comment by CrossR** [comment]
I've currently been running 

```json
"terminal.integrated.shell.osx": "/usr/local/bin/bash",
"terminal.integrated.shellArgs.osx": ["--login"],
```

(Change the terminal path to whatever term you use, mine is for bash from brew, but zsh and fish should work the exact same, just copy the output of `echo $SHELL`).

This causes the following to happen (or equivalent for your shell):

> When bash is invoked as an interactive login shell, or as a non-interactive shell with the --login option, it first reads and executes commands from the file /etc/profile, if that file exists. After reading that file, it looks for ~/.bash_profile, ~/.bash_login, and ~/.profile, in that order, and reads and executes commands from the first one that exists and is readable.

Maybe someone who knows the shell better than I do will know if this is fine to do, or if it could cause issues in the future. If its fine, this is a reasonable "fix" for now, and we can see if there is a better way to achieve the same env loading or not. I've been using it for a week or so with no issues.

**Comment by bryphe** [comment]
This should be fixed now in #2634 :+1:
